app/utils/gallery/index.test.ts

```diff
@@ -0,0 +1,283 @@
+jest.mock('react-native-reanimated', () => ({
+    measure: jest.fn(() => ({
+        pageX: 100,
+        pageY: 200,
+        width: 300,
+        height: 400,
+    })),
+}));
+
+describe('Gallery utils', () => {
+    afterAll(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('clamp', () => {
+        it('should clamp a value within the given bounds', () => {
+            expect(clamp(5, 1, 10)).toBe(5);
+            expect(clamp(-5, 0, 10)).toBe(0);
+            expect(clamp(15, 0, 10)).toBe(10);
+        });
+    });
+
+    describe('clampVelocity', () => {
+        it('should clamp positive velocity within the given bounds', () => {
+            expect(clampVelocity(5, 1, 10)).toBe(5);
+            expect(clampVelocity(0.5, 1, 10)).toBe(1);
+            expect(clampVelocity(15, 1, 10)).toBe(10);
+        });
+
+        it('should clamp negative velocity within the given bounds', () => {
+            expect(clampVelocity(-5, 1, 10)).toBe(-5);
+            expect(clampVelocity(-0.5, 1, 10)).toBe(-1);
+            expect(clampVelocity(-15, 1, 10)).toBe(-10);
+        });
+    });
+
+    describe('fileToGalleryItem', () => {
+        it('should convert file info to gallery item with image type', () => {
+            const file = {
+                extension: 'jpg',
+                height: 600,
+                id: '123',
+                mime_type: 'image/jpeg',
+                name: 'test-image',
+                post_id: 'post123',
+                size: 5000,
+                localPath: '/path/to/image.jpg',
+                width: 800,
+            } as FileInfo;
+            const result = fileToGalleryItem(file);
+            expect(result.type).toBe('image');
+            expect(result.uri).toBe(file.localPath);
+        });
+
+        it('should convert file info to gallery item with video type', () => {
+            const file = {
+                extension: 'mp4',
+                height: 600,
+                id: '123',
+                mime_type: 'video/mp4',
+                name: 'test-video',
+                post_id: 'post123',
+                size: 10000,
+                localPath: '/path/to/video.mp4',
+                mini_preview: '/path/to/preview.jpg',
+                width: 800,
+            } as FileInfo;
+            const result = fileToGalleryItem(file);
+            expect(result.type).toBe('video');
+            expect(result.posterUri).toBe(file.mini_preview);
+        });
+
+        it('should convert file info to gallery item with video type and assign a file id that starts with uid', () => {
+            const file = {
+                extension: 'mp4',
+                height: 600,
+                mime_type: 'video/mp4',
+                name: 'test-video',
+                post_id: 'post123',
+                size: 10000,
+                localPath: '/path/to/video.mp4',
+                mini_preview: '/path/to/preview.jpg',
+                width: 800,
```

Comment:
nit: this file object is duplicated with object on line 113-122 (with very few difference).  I typically am okay with having duplicates in test so that devs can easily follow, but in this instance, extracting some of the object keys (and values) out may make it a lot easier to see what key-value really matters within the test suites.

```
    describe('fileToGalleryItem', () => {

        const videoFileInfo = {
            extension: 'mp4',
            height: 600,
            id: '123',
            mime_type: 'video/mp4',
            name: 'test-video',
            post_id: 'post123',
            size: 10000,
            localPath: '/path/to/video.mp4',
            width: 800,
        }

        ...

        it('should convert file info to gallery item with video type with correct posterUri', () => {
            const file = {
                ...videoFileInfo,
                
                // easier to see that mini_preview is what causing posterUri
                mini_preview: '/path/to/preview.jpg', 
            } as FileInfo;

            const result = fileToGalleryItem(file);
            expect(result.type).toBe('video');
            expect(result.posterUri).toBe(file.mini_preview);
        }

        it('should convert file info to gallery item and assign a file id that starts with uid', () => {

            // may help understand that absence of id is what causing result.id to have a generated uid
            delete videoFileInfo['id']; 

            const result = fileToGalleryItem(videoFileInfo);
            expect(result.id.startsWith('uid')).toBeTruthy();
        }
    }
```

Reply:
Would there be any benefit to check for non-`image` and non-`video`?

```
    const fileInfo = {
        extension: 'csv',
        mime_type: 'text/csv',
        ...
    }

    // it would return type: file

    expect(result.type).toBe('file');
```